documentation/using-azure-identity.md

```diff
@@ -0,0 +1,325 @@
+to check if you have AAD tenant already or, if not, create one.
+
+Once you have a tenant, you can create an app registration by following [this
+quickstart guide for app
+registrations](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
+Your app registration holds the configuration for how your application will
+authenticate users and services, so it's very important to it set up correctly
+before using any of the credential types below.  The section on each credential
+will indicate which configuration settings it needs and how to use them.
+
+### Should my App be Single or Multi Tenant?
+
+One decision you will need to make up front when registering your app is whether
+it will be single or multi-tenant, and more importantly, if the multi-tenant app
+registration also supports personal Microsoft accounts. The primary deciding
+factor is whether your application will be used only by users and services
+inside of your AAD tenant or if you'd like other organizations and individuals
+to use it.
+
+The [app registration quickstart
+guide](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#register-a-new-application-using-the-azure-portal)
+gives a helpful breakdown for the various tenancy options in the "Supported
+account types" documentation.
+
+You can easily change your application from single to multi-tenant and vice
+versa after it is created, but you cannot currently change it to support
+personal Microsoft accounts after it's already created.
+
+## Understanding the Credential Types
+
+Microsoft identity platform provides a variety of authentication flows that
+serve different use cases and application types.  A primary differentiator
+between these flows is whether the "client" that initiates the flow is running
+on a user device or on a system managed by the application developer (like a web
+server).  The [Microsoft Authentication
+Library](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-client-applications)
+documentation describes this distinction as _public_ versus _confidential_
+clients.
+
+Most of the credential types are strictly public or confidential as they serve a
+specific purpose, like authenticating a backend service for use with storage
+APIs.  Some credentials may be both public or confidential depending on how you
+configure them.  For example, the [authorization code
+flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow)
+can be initiated from a mobile application _or_ from within a web application
+running in a server.
+
+## Choosing a Credential Type
+
+Here's a high level decision tree to help with choosing an appropriate
+credential for your application:
+
+- **Is the application deployed to a server?**
+
+  - **Do you want to avoid entering credentials directly in your code or
```

Comment:
Is this really the decision point?  Naively, I would assume that if the service I am using to run my application supports managed identity, I should use managed identity and that `EnvironmentCredential` is only really used in cases where managed identity could not be used?   Could be wrong here.

Reply:
I agree that this should be more a question of is Managed Identity available in the environment I'm deploying to. Also I think that we should highlight the DefaultAzureCredential as the first choice in this case.

Reply:
Thanks for the feedback, you're both right, I'll reshape the decision tree to emphasize `DefaultAzureCredential` and `ManagedIdentityCredential`